js/shared.js

```diff
@@ -378,4 +377,75 @@ async function getAllWitnessesOfSource(source){
  */ 
 function undoBrowserSelection(s){
     s?.removeAllRanges?.() ?? s?.empty?.()
+}
+
+/** 
+ * Confirm the shelfmark's inclusion in "GoG-manuscripts" dynamic collection. 
+ * @see {@link findShelfmark #113} to verify shelfmark is appropriate. 
+ *
+ * @async
+ * @function addManuscriptToGoG
+ * @param { string } shelfmark Identifier to include.
+ */ 
+async function addManuscriptToGoG(shelfmark) {
+    try {
+        /** Wash shelfmark by 
+         * Removing specific special characters @ $ % * ?
+         * Replace multiple spaces with a single space
+         * Removing trailing or leading whitespace
+         * */
+        const cleanShelfmark = shelfmark.replace(/[@$%*?]+/g, '') 
+        .replace(/\s+/g, ' ') 
+        .trim() 
+
+        // check for existing annotation with cleaned shelfmark
+        const query = {
+            "target": `${cleanShelfmark}`,
+        }
+        const response = await fetch(`${__constants.tiny+"/query"}`, {
```

Comment:
Change this around a little bit so you end up with the Array you want or your handled error if things go wrong  You can control what this variable becomes by returning something in the caught error

`const existingAnnotations = fetch().then(resp=>resp.json()).catch(err => {return new Error("AHHH")})`